agent/proxyprocess/proxy.go

```diff
@@ -1,16 +1,9 @@
-// Package proxy contains logic for agent interaction with proxies,
-// primarily "managed" proxies. Managed proxies are proxy processes for
-// Connect-compatible endpoints that Consul owns and controls the lifecycle
-// for.
+// Package proxy contains logic for agent interaction with proxies.
 //
 // This package does not contain the built-in proxy for Connect. The source
 // for that is available in the "connect/proxy" package.
 package proxyprocess
 
-import (
-	"github.com/hashicorp/consul/agent/structs"
-)
-
 const (
```

Comment:
I think after you nuke what's already in the PR the only legit references will be 

```
$ git grep "proxyprocess"
...(omitted irrelevant lines)...
command/connect/envoy/envoy.go: proxyAgent "github.com/hashicorp/consul/agent/proxyprocess"
command/connect/proxy/proxy.go: proxyAgent "github.com/hashicorp/consul/agent/proxyprocess"

$ git grep proxyAgent
command/connect/envoy/envoy.go: proxyAgent "github.com/hashicorp/consul/agent/proxyprocess"
command/connect/envoy/envoy.go:         c.proxyID = os.Getenv(proxyAgent.EnvProxyID)
command/connect/envoy/envoy.go:         c.sidecarFor = os.Getenv(proxyAgent.EnvSidecarFor)
command/connect/envoy/envoy.go:         if proxyToken := os.Getenv(proxyAgent.EnvProxyToken); proxyToken != "" {
command/connect/proxy/proxy.go: proxyAgent "github.com/hashicorp/consul/agent/proxyprocess"
command/connect/proxy/proxy.go:         c.proxyID = os.Getenv(proxyAgent.EnvProxyID)
command/connect/proxy/proxy.go:         c.sidecarFor = os.Getenv(proxyAgent.EnvSidecarFor)
command/connect/proxy/proxy.go:         c.http.SetToken(os.Getenv(proxyAgent.EnvProxyToken))
```

Which leads me to believe that maybe those constant env var names should just be moved to some place like `command/connect/env.go` or something and then you can nuke the whole `proxyprocess` package.